ramanujan/cmf/cmf.py

```diff
@@ -22,45 +22,71 @@ def __init__(self, matrices: Dict[sp.Symbol, Matrix]):
         """
         Initializes a CMF with `Mx` and `My` matrices
         """
-
         self.matrices = matrices
-
+        self.assert_conserving()
         assert (
             n not in self.matrices.keys()
         ), "Do not use symbol n as an axis, it's reserved for PCF conversions"
 
-        for x, y in itertools.combinations(self.matrices.keys(), 2):
-            Mx = self.matrices[x]
-            My = self.matrices[y]
-            Mxy = simplify(Mx * My.subs({x: x + 1}))
-            Myx = simplify(My * Mx.subs({y: y + 1}))
-            if simplify(Mxy - Myx) != Matrix([[0, 0], [0, 0]]):
-                raise ValueError(f"M{x} and M{y} matrices are not conserving!")
-
     def __eq__(self, other) -> bool:
         return self.matrices == other.matrices
 
     def __repr__(self) -> str:
         return f"CMF({self.matrices})"
 
-    def M(self, axis: sp.Symbol) -> Matrix:
+    def are_conserving(
+        self,
+        x: sp.Symbol,
+        y: sp.Symbol,
+        x_forward: bool = True,
+        y_forward: bool = True,
+    ) -> bool:
+        Mx = self.M(x, x_forward)
+        My = self.M(y, y_forward)
+        Mxy = simplify(Mx * My.subs({x: x + 1 if x_forward else x - 1}))
+        Myx = simplify(My * Mx.subs({y: y + 1 if y_forward else y - 1}))
+        return Mxy == Myx
+
+    def assert_conserving(self, check_negatives: bool = False) -> None:
+        for x, y in itertools.combinations(self.matrices.keys(), 2):
+            if not self.are_conserving(x, y, True, True):
+                raise ValueError(f"M({x}) and M({y}) matrices are not conserving!")
+
+            if check_negatives:
+                if not self.are_conserving(x, y, False, True):
+                    raise ValueError(f"M(-{x}) and M({y}) matrices are not conserving!")
+                if not self.are_conserving(x, y, True, False):
+                    raise ValueError(f"M({x}) and M(-{y}) matrices are not conserving!")
+                if not self.are_conserving(x, y, False, False):
+                    raise ValueError(
+                        f"M(-{x}) and M(-{y}) matrices are not conserving!"
+                    )
+
+    def M(self, axis: sp.Symbol, sign: bool = True) -> Matrix:
         """
         Returns the axis matrix for a given axis.
+
```

Comment:
Document why normalization here is dangerous!
(this as it might normalize by axis parameters, which will make it non-conservative)